packages/table/preview/mutableTable.tsx

```diff
@@ -0,0 +1,969 @@
+import { IRegion } from "../src/regions";
+import { DenseGridMutableStore } from "./denseGridMutableStore";
+import { LocalStore } from "./localStore";
+
+export enum FocusStyle {
+    TAB,
+    TAB_OR_CLICK,
+}
+
+export enum CellContent {
+    EMPTY,
+    CELL_NAMES,
+    LONG_TEXT,
+}
+
+type IMutableStateUpdateCallback = (
+    stateKey: keyof IMutableTableState,
+) => ((event: React.FormEvent<HTMLElement>) => void);
+
+const COLUMN_COUNTS = [0, 1, 5, 20, 100, 1000];
+
+const ROW_COUNTS = [0, 1, 5, 20, 100, 1000, 100000];
+
+const FROZEN_COLUMN_COUNTS = [0, 1, 2, 5, 20, 100, 1000];
+const FROZEN_ROW_COUNTS = [0, 1, 2, 5, 20, 100, 1000];
+
+const REGION_CARDINALITIES = [
+    RegionCardinality.CELLS,
+    RegionCardinality.FULL_ROWS,
+    RegionCardinality.FULL_COLUMNS,
+    RegionCardinality.FULL_TABLE,
+];
+
+const CELL_CONTENTS = [CellContent.EMPTY, CellContent.CELL_NAMES, CellContent.LONG_TEXT];
+
+const TRUNCATED_POPOVER_MODES: TruncatedPopoverMode[] = [
+    TruncatedPopoverMode.ALWAYS,
+    TruncatedPopoverMode.NEVER,
+    TruncatedPopoverMode.WHEN_TRUNCATED,
+];
+
+const COLUMN_COUNT_DEFAULT_INDEX = 3;
+const ROW_COUNT_DEFAULT_INDEX = 4;
+
+const FROZEN_COLUMN_COUNT_DEFAULT_INDEX = 0;
+const FROZEN_ROW_COUNT_DEFAULT_INDEX = 0;
+
+const LONG_TEXT_MIN_LENGTH = 5;
+const LONG_TEXT_MAX_LENGTH = 40;
+const ALPHANUMERIC_CHARS = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+const CELL_CONTENT_GENERATORS = {
+    [CellContent.CELL_NAMES]: Utils.toBase26CellName,
+    [CellContent.EMPTY]: () => "",
+    [CellContent.LONG_TEXT]: () => {
+        const randomLength = getRandomInteger(LONG_TEXT_MIN_LENGTH, LONG_TEXT_MAX_LENGTH);
+        return Utils.times(randomLength, () => {
+            const randomIndex = getRandomInteger(0, ALPHANUMERIC_CHARS.length - 1);
+            return ALPHANUMERIC_CHARS[randomIndex];
+        }).join("");
+    },
+};
+
+// TODO: Pull these from @blueprintjs/docs
+
+/** Event handler that exposes the target element's value as a boolean. */
+function handleBooleanChange(handler: (checked: boolean) => void) {
+    return (event: React.FormEvent<HTMLElement>) => handler((event.target as HTMLInputElement).checked);
+}
+
+/** Event handler that exposes the target element's value as a string. */
+function handleStringChange(handler: (value: string) => void) {
+    return (event: React.FormEvent<HTMLElement>) => handler((event.target as HTMLInputElement).value);
+}
+
+/** Event handler that exposes the target element's value as a number. */
+function handleNumberChange(handler: (value: number) => void) {
+    return handleStringChange(value => handler(+value));
+}
+
+function getRandomInteger(min: number, max: number) {
+    // min and max are inclusive
+    return Math.floor(min + Math.random() * (max - min + 1));
+}
+
+function contains(arr: any[], value: any) {
+    return arr.indexOf(value) >= 0;
+}
+
+const DEFAULT_STATE: IMutableTableState = {
```

Comment:
Can we define this _below_ the `IMutableTableState` interface? Keeps the `DEFAULTS` definition closer to its usage in the constructor, and that order would just make more sense to me.